esp-hal/src/touch.rs

```diff
@@ -0,0 +1,596 @@
+//! # Capacitive Touch Sensor
+//!
+//! ## Overview
+//!
+//! The touch sensor peripheral allows for cheap and robust user interfaces by
+//! e.g., dedicating a part of the pcb as touch button.
+//!
+//! ## Examples
+//!
+//! ```rust, no_run
+#![doc = crate::before_snippet!()]
+//! # use esp_hal::touch::{Touch, TouchPad};
+//! # use esp_hal::gpio::Io;
+//! let io = Io::new(peripherals.GPIO, peripherals.IO_MUX);
+//! let touch_pin0 = io.pins.gpio2;
+//! let touch = Touch::continous_mode(peripherals.TOUCH, None);
+//! let mut touchpad = TouchPad::new(touch_pin0, &touch);
+//! // ... give the peripheral some time for the measurement
+//! let touch_val = touchpad.read();
+//! # }
+//! ```
+//! 
+//! ## Implementation State:
+//!
+//! Mostly feature complete, missing:
+//!
+//! - Touch sensor slope control
+//! - Deep Sleep support (wakeup from Deep Sleep)
+
+#![deny(missing_docs)]
+
+use core::marker::PhantomData;
+
+use crate::{
+    gpio::TouchPin,
+    peripheral::{Peripheral, PeripheralRef},
+    peripherals::{RTC_CNTL, SENS, TOUCH},
+    private::{Internal, Sealed},
+    Blocking,
+    Mode,
+};
+#[cfg(feature = "async")]
+use crate::{rtc_cntl::Rtc, Async, InterruptConfigurable};
+
+/// A marker trait describing the mode the touch pad is set to.
+pub trait TouchMode: Sealed {}
+
+/// Marker struct for the touch peripherals manual trigger mode. In the
+/// technical reference manual, this is referred to as "start FSM via software".
+#[derive(Debug)]
```

Comment:
nitpick but when deriving `Debug` we should also optionally derive `defmt::Format` like this `#[cfg_attr(feature = "defmt", derive(defmt::Format))]` (also for the other structs)

Reply:
Good point. Added this.